packages/web/src/common/store/wallet/sagas.ts

```diff
@@ -179,54 +179,58 @@ function* fetchBalanceAsync() {
-  // is frozen due to a recent optimistic update
-  const isBalanceFrozen = yield* call(getIsBalanceFrozen)
-  if (isBalanceFrozen) return
-
-  const localBalanceChange: ReturnType<typeof getLocalBalanceDidChange> =
-    yield* select(getLocalBalanceDidChange)
-
-  const [currentEthAudioWeiBalance, currentSolAudioWeiBalance] = yield* all([
-    call(
-      [walletClient, 'getCurrentBalance'],
-      /* bustCache */ localBalanceChange
-    ),
-    call([walletClient, 'getCurrentWAudioBalance'])
-  ])
+  try {
+    // Opt out of balance refreshes if the balance
+    // is frozen due to a recent optimistic update
+    const isBalanceFrozen = yield* call(getIsBalanceFrozen)
+    if (isBalanceFrozen) return
 
-  const associatedWalletBalance: BNWei = yield* call(
-    [walletClient, 'getAssociatedWalletBalance'],
-    account.user_id,
-    /* bustCache */ localBalanceChange
-  )
+    const localBalanceChange: ReturnType<typeof getLocalBalanceDidChange> =
+      yield* select(getLocalBalanceDidChange)
 
-  const audioWeiBalance = currentEthAudioWeiBalance.add(
-    currentSolAudioWeiBalance
-  ) as BNWei
+    const [currentEthAudioWeiBalance, currentSolAudioWeiBalance] = yield* all([
+      call(
+        [walletClient, 'getCurrentBalance'],
+        /* bustCache */ localBalanceChange
+      ),
+      call([walletClient, 'getCurrentWAudioBalance'])
+    ])
 
-  const useSolAudio = yield* call(
-    getFeatureEnabled,
-    FeatureFlags.ENABLE_SPL_AUDIO
-  )
-  if (useSolAudio) {
-    const totalBalance = audioWeiBalance.add(associatedWalletBalance) as BNWei
-    yield* put(
-      setBalance({
-        balance: weiToString(audioWeiBalance),
-        totalBalance: weiToString(totalBalance)
-      })
+    const associatedWalletBalance: BNWei = yield* call(
+      [walletClient, 'getAssociatedWalletBalance'],
+      account.user_id,
+      /* bustCache */ localBalanceChange
     )
-  } else {
-    const totalBalance = currentEthAudioWeiBalance.add(
-      associatedWalletBalance
+
+    const audioWeiBalance = currentEthAudioWeiBalance.add(
+      currentSolAudioWeiBalance
     ) as BNWei
-    yield* put(
-      setBalance({
-        balance: weiToString(currentEthAudioWeiBalance),
-        totalBalance: weiToString(totalBalance)
-      })
+
+    const useSolAudio = yield* call(
+      getFeatureEnabled,
+      FeatureFlags.ENABLE_SPL_AUDIO
     )
+    if (useSolAudio) {
+      const totalBalance = audioWeiBalance.add(associatedWalletBalance) as BNWei
+      yield* put(
+        setBalance({
+          balance: weiToString(audioWeiBalance),
+          totalBalance: weiToString(totalBalance)
+        })
+      )
+    } else {
+      const totalBalance = currentEthAudioWeiBalance.add(
+        associatedWalletBalance
+      ) as BNWei
+      yield* put(
+        setBalance({
+          balance: weiToString(currentEthAudioWeiBalance),
+          totalBalance: weiToString(totalBalance)
+        })
+      )
+    }
+  } catch (err) {
+    console.error(err)
```

Comment:
Maybe also trigger our `handleError` sentry stuff here like this:
https://github.com/AudiusProject/audius-client/blob/370a7cabdf901f3c805fba598866acf94ec1ee6b/packages/web/src/store/application/ui/buy-audio/sagas.ts#L926-L930